core/client/hammer/report.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hammer
+
+import (
+	"log"
+	"sort"
+	"time"
+)
+
+// Result represents the output of one operation.
+type Result struct {
+	Err        error
+	Start, End time.Time
+}
+
+// Stats hold statistics on the output of many operations.
+type Stats struct {
+	RequestCount int64
+	AvgTotal     time.Duration // Sum of individual latencies.
+	Total        time.Duration // Total wall time.
+	Fastest      time.Duration
+	Slowest      time.Duration
+	Average      time.Duration // Sum of latencies / count
+	QPS          float64       // Wall time / count
+	ErrorDist    map[string]int
+}
+
+func collectStats(results <-chan Result) *Stats {
+	s := &Stats{
+		ErrorDist: make(map[string]int),
+	}
+	st := time.Now()
+	for res := range results {
+		if res.Err != nil {
+			s.ErrorDist[res.Err.Error()]++
+			continue
+		}
+
+		s.RequestCount++
```

Comment:
To me `RequestCount` looks more like a total #Ok+#Err counter rather than just #Ok. Rename to `OkCount`?

Reply:
Deleted file